### PR TITLE
VKC-67: Volume Expansion: OFFLINE and ONLINE expansion capability

### LINKED ISSUE #27: Volume Expansion
### Describe the bug

I have created a storage class:

```
apiVersion: v1
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  annotations:
    storageclass.kubernetes.io/is-default-class: "true"
  name: pure
provisioner: named-disk.csi.cloud-director.vmware.com
reclaimPolicy: Delete
parameters:
  storageProfile: "Pure"
  filesystem: "ext4"
allowVolumeExpansion: true
```

I can create a PVC: 

```
NAME   STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
test   Bound    pvc-acd8f9a9-dcd3-4735-9888-7d2d9748c659   1Gi        RWO            pure           12m
```

However if I edit the PVC and change spec.resources.requests.storage to a new size, e.g. 2Gi, the change doesn't happen and the following is logged:

```
Events:
  Type     Reason                 Age                   From                                                                                                      Message
  ----     ------                 ----                  ----                                                                                                      -------
  Normal   Provisioning           8m5s                  named-disk.csi.cloud-director.vmware.com_csi-vcd-controllerplugin-0_097b7dca-fddc-4972-bcc2-b0b6e9d30b62  External provisioner is provisioning volume for claim "default/test"
  Normal   ExternalProvisioning   7m54s (x4 over 8m5s)  persistentvolume-controller                                                                               waiting for a volume to be created, either by external provisioner "named-disk.csi.cloud-director.vmware.com" or manually created by system administrator
  Normal   ProvisioningSucceeded  7m47s                 named-disk.csi.cloud-director.vmware.com_csi-vcd-controllerplugin-0_097b7dca-fddc-4972-bcc2-b0b6e9d30b62  Successfully provisioned volume pvc-acd8f9a9-dcd3-4735-9888-7d2d9748c659
  Warning  ExternalExpanding      7m11s                 volume_expand                                                                                             Ignoring the PVC: didn't find a plugin capable of expanding the volume; waiting for an external controller to process this PVC.
```

I can navigate to Datacenter -> Storage -> Named Disks in the vCD UI -> select the disk -> edit -> change the size and this is successful in vCenter. But the PVC is never updated.

### Reproduction steps

```bash
1. Create PVC
2. Edit PVC and change size - logs that plugin not capable
3. Edit PVC via vCD - not updated in k8s
```

### Expected behavior

It should be possible to expand the disk by editing the PVC. 
Alternatively if the PVC is expanded via vCD, this should be updated in the k8s PVC.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by arunmk**
## Description
Please provide a brief description of the changes proposed in this Pull Request
- Added Offline Volume expansion.
- This also needs a CSI Resizer plugin component to be added. Updated all requirements for air-gapped to the extent of my understanding.

## Checklist
- [X] tested locally
- [X] updated any relevant dependencies
- [X] updated any relevant documentation or examples

## Testing Done
Please provide a screenshot of the testing results for the code change in this Pull Request. Verify that this pull request's code change will not affect CSI's normal operation
This has been tested with an addition to the storageClass (as described in the sample storageclass change. What was done was:
1. create new storage-class with expansion allowed
2. Create a PVC
3. Increase size of PVC and apply it
4. Wait for storage to expand (offline expansion)
5. Make a pod consume the storage
6. Increase size of PVC and apply it
7. Note that the storage does not expand (online not allowed)
8. Delete the pod
9. Wait for storage to expand as required (since it is offline)
10. Reduce storage and apply: this should fail from the k8s validation itself.

This also includes automated tests.

## Issue
Fixes #27

- - -
This change is [Reviewable](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144)

### CODE REVIEW

**Review by ymo24 — COMMENTED** [review]
> Reviewed 9 of 16 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144)* status: 9 of 16 files reviewed, 1 unresolved discussion (waiting on @arunmk and @rocknes)

___
*[`manifests/csi-node.yaml` line 93 at r1](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-NOpW5TlA2eVzP2hsPuz:-NOpW5Tm-yAvo-hFAwTr:b-5yiq5w) ([raw file](https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/683e6e68f89e42d09d7fee72438b62fbb040cd4b/manifests/csi-node.yaml#L93)):*
> ```YAML
>               add: ["SYS_ADMIN"]
>             allowPrivilegeEscalation: true
>           image: harbor-repo.vmware.com/vcloud/cloud-director-named-disk-csi-driver:ak-expansion-6.d1c9bc2
> ```

Should we use main-branch.xxxxx here?

**Review by ymo24 — APPROVED** [review]
:lgtm: 

> *[Reviewable](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-:-NOpX_ZRAXhngweKvT9V:bg7igkv)* status: 9 of 16 files reviewed, 1 unresolved discussion (waiting on @arunmk and @rocknes)

**Review by ymo24 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144)* status: 9 of 16 files reviewed, 2 unresolved discussions (waiting on @arunmk and @rocknes)

___
*[`manifests/csi-node-crs.yaml` line 93 at r1](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-NOrLclRAOZLn5rzyO3a:-NOrLclRAOZLn5rzyO3b:b-2guby3) ([raw file](https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/683e6e68f89e42d09d7fee72438b62fbb040cd4b/manifests/csi-node-crs.yaml#L93)):*
> ```YAML
>               add: ["SYS_ADMIN"]
>             allowPrivilegeEscalation: true
>           image: harbor-repo.vmware.com/vcloud/cloud-director-named-disk-csi-driver:ak-expansion-6.d1c9bc2
> ```

same here

**Comment by arunmk** [comment]
*[`manifests/csi-node.yaml` line 93 at r1](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-NOpW5TlA2eVzP2hsPuz:-NOrxazpA08fnq7o-jkZ:b-ofbqcg) ([raw file](https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/683e6e68f89e42d09d7fee72438b62fbb040cd4b/manifests/csi-node.yaml#L93)):*
<details><summary><i>Previously, ymo24 wrote…</i></summary><blockquote>

Should we use main-branch.xxxxx here?
</blockquote></details>

Thanks Yan. Fixed.

**Review by arunmk — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144)* status: 7 of 16 files reviewed, 2 unresolved discussions (waiting on @rocknes and @ymo24)

___
*[`manifests/csi-node-crs.yaml` line 93 at r1](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-NOrLclRAOZLn5rzyO3a:-NOrxeh20j3XX0f5_Q4f:b-896fix) ([raw file](https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/683e6e68f89e42d09d7fee72438b62fbb040cd4b/manifests/csi-node-crs.yaml#L93)):*
<details><summary><i>Previously, ymo24 wrote…</i></summary><blockquote>

same here
</blockquote></details>

Done.

**Comment by ekarlso** [comment]
Seems `csi-vcd-node-sa` is missing in RBAC to list PV and PVC btw after testing this locally. Can you please check ? :)

**Comment by mate201** [comment]
It will be nice to have this feature.

**Comment by erSitzt** [comment]
Whats the holdup ? :)

**Review by Anirudh9794 — COMMENTED** [review]
> Reviewed 18 of 1656 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144)* status: 19 of 1657 files reviewed, 9 unresolved discussions (waiting on @arunmk, @rocknes, and @ymo24)

___
*[`manifests/csi-controller-crs.yaml.template` line 175 at r3](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-NnQjMWABzyuGe-R6WqK:-NnQjMWABzyuGe-R6WqL:b-h1uvlp) ([raw file](https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/5b51a33652fab0eae869d59ea9c49e08f37870bc/manifests/csi-controller-crs.yaml.template#L175)):*
> ```
>               mountPath: /var/lib/csi/sockets/pluginproxy/
>         - name: csi-resizer
>           image: k8s.gcr.io/sig-storage/csi-resizer:v1.4.0
> ```

I think this should be `__REGISTRY__/sig-storage/csi-resizer:v1.4.0`
___
*[`manifests/csi-controller.yaml.template` line 175 at r3](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-NnQjWCr2wQz11qtdYIx:-NnQjWCr2wQz11qtdYIy:brewamo) ([raw file](https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/5b51a33652fab0eae869d59ea9c49e08f37870bc/manifests/csi-controller.yaml.template#L175)):*
> ```
>               mountPath: /var/lib/csi/sockets/pluginproxy/
>         - name: csi-resizer
>           image: k8s.gcr.io/sig-storage/csi-resizer:v1.4.0
> ```

This image should have `__REGISTRY__`
___
*[`manifests/csi-node-crs.yaml.template` line 76 at r3](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-NnQjieEAl6vnIol-ZsF:-NnQjieEAl6vnIol-ZsG:bqdzvu2) ([raw file](https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/5b51a33652fab0eae869d59ea9c49e08f37870bc/manifests/csi-node-crs.yaml.template#L76)):*
> ```
>               mountPath: /registration
>         - name: csi-resizer
>           image: k8s.gcr.io/sig-storage/csi-resizer:v1.4.0
> ```

__REGISTRY__ here as well
___
*[`manifests/csi-node.yaml.template` line 77 at r3](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-NnQjoIM0zpyFclFRNAS:-NnQjoIM0zpyFclFRNAT:bqdzvu2) ([raw file](https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/5b51a33652fab0eae869d59ea9c49e08f37870bc/manifests/csi-node.yaml.template#L77)):*
> ```
>               mountPath: /registration
>         - name: csi-resizer
>           image: k8s.gcr.io/sig-storage/csi-resizer:v1.4.0
> ```

__REGISTRY__ here as well
___
*[`manifests/vcloud-csi-config.yaml.bak` line 4 at r3](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-NnQjsxCAzjbyJ3Ge3E9:-NnQjsxCAzjbyJ3Ge3EA:bxqxpwc) ([raw file](https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/5b51a33652fab0eae869d59ea9c49e08f37870bc/manifests/vcloud-csi-config.yaml.bak#L4)):*
> ```
> apiVersion: v1
> kind: ConfigMap
> metadata:
> ```

I think this file was committed by mistake
___
*[`pkg/csi/controller.go` line 295 at r3](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-NnQmenr9qOerAxWRAv-:-NnQmenr9qOerAxWRAv0:b-x790d4) ([raw file](https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/48314b27ba85aff74d25569a149ce5db5d10ee6e/pkg/csi/controller.go#L295)):*
> ```Go
> 	if len(nodeID) == 0 {
> 		return nil, status.Errorf(codes.InvalidArgument,
> 			"ControllerUnpublishVolume: Node ID must be provided")
> ```

Just wanted to understand why this check was not required. Is it something like if no 
___
*[`pkg/csi/controller.go` line 410 at r3](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-NnQo7vx09f1pAIawmir:-NnQo7vx09f1pAIawmis:bvlt3o) ([raw file](https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/5b51a33652fab0eae869d59ea9c49e08f37870bc/pkg/csi/controller.go#L410)):*
> ```Go
> 	}
> 
> 	if req.CapacityRange.LimitBytes > 0 && req.CapacityRange.RequiredBytes > req.CapacityRange.LimitBytes {
> ```

How can one set the limit bytes in the request?

File: manifests/csi-controller.yaml [hunk 1]
```diff
@@ -134,6 +171,19 @@ spec:
           volumeMounts:
             - name: socket-dir
               mountPath: /var/lib/csi/sockets/pluginproxy/
+        - name: csi-resizer
+          image: k8s.gcr.io/sig-storage/csi-resizer:v1.4.0
```
Comment (rocknes) [inline manifests/csi-controller.yaml:175]: Shouldn't this be k8s.registry.io?

**Review by ymo24 — APPROVED** [review]
> Reviewed 1656 of 1656 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144)* status: all files reviewed, 8 unresolved discussions (waiting on @Anirudh9794 and @arunmk)

___
*[`manifests/csi-controller-crs.yaml.template` line 175 at r3](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-NnQjMWABzyuGe-R6WqK:-Nnc1BDQ9lXtX3fOTlFR:b32v57) ([raw file](https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/5b51a33652fab0eae869d59ea9c49e08f37870bc/manifests/csi-controller-crs.yaml.template#L175)):*
<details><summary><i>Previously, Anirudh9794 wrote…</i></summary><blockquote>

I think this should be `__REGISTRY__/sig-storage/csi-resizer:v1.4.0`
</blockquote></details>

+1

**Review by Anirudh9794 — APPROVED** [review]
:lgtm: with some minor comments

> Reviewed 1 of 16 files at r1, 14 of 1656 files at r3.
> *[Reviewable](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-:-Nnf-_pU2LY36Cl6saTl:b-v72ds0)* status: all files reviewed, 10 unresolved discussions (waiting on @arunmk and @ymo24)

___
*[`pkg/csi/controller.go` line 471 at r3](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-NnekWt77J35aAswGnZ0:-NnekWt77J35aAswGnZ1:bvzpqj7) ([raw file](https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/5b51a33652fab0eae869d59ea9c49e08f37870bc/pkg/csi/controller.go#L471)):*
> ```Go
> 			// an attached VM. Then it will use the ONLINE route using the VM API.
> 			expansionMode = csi.PluginCapability_VolumeExpansion_OFFLINE
> 		}
> ```

do we need to have an else condition within the block
```
if volumeAccessMode.Mode != csi.VolumeCapability_AccessMode_MULTI_NODE_MULTI_WRITER &&
		volumeCapability.GetBlock() == nil {
}
```
___
*[`pkg/csi/controller.go` line 560 at r3](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-Nnel5ySBPkmVzHiKKMW:-Nnel5ySBPkmVzHiKKMX:b-pdome0) ([raw file](https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/5b51a33652fab0eae869d59ea9c49e08f37870bc/pkg/csi/controller.go#L560)):*
> ```Go
> 			NodeExpansionRequired: true,
> 		}, nil
> 		break
> ```

break seems to be redundant here

File: manifests/csi-controller.yaml (hunk 1 above)
Comment (arunmk) [reply manifests/csi-controller.yaml:175]: Modified in all files.

**Comment by arunmk** [comment]
*[`manifests/vcloud-csi-config.yaml.bak` line 4 at r3](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-NnQjsxCAzjbyJ3Ge3E9:-NngXZ3uA9UFNZIPinVg:b-sg81zk) ([raw file](https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/5b51a33652fab0eae869d59ea9c49e08f37870bc/manifests/vcloud-csi-config.yaml.bak#L4)):*
<details><summary><i>Previously, Anirudh9794 wrote…</i></summary><blockquote>

I think this file was committed by mistake
</blockquote></details>

Thanks, removed.

**Comment by arunmk** [comment]
*[`pkg/csi/controller.go` line 471 at r3](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-NnekWt77J35aAswGnZ0:-NngY3IQ1BeHyFR08q5s:b-2jg14g) ([raw file](https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/5b51a33652fab0eae869d59ea9c49e08f37870bc/pkg/csi/controller.go#L471)):*
<details><summary><i>Previously, Anirudh9794 wrote…</i></summary><blockquote>

do we need to have an else condition within the block
```
if volumeAccessMode.Mode != csi.VolumeCapability_AccessMode_MULTI_NODE_MULTI_WRITER &&
		volumeCapability.GetBlock() == nil {
}
```
</blockquote></details>

The default sets `expansionMode = csi.PluginCapability_VolumeExpansion_ONLINE` and then there is a `if` block. So the `else is the default itself.

Am I understanding your comment correctly @Anirudh9794 ?

**Comment by arunmk** [comment]
*[`pkg/csi/controller.go` line 560 at r3](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-Nnel5ySBPkmVzHiKKMW:-NngYMmiBFNwEQTs5_jA:b-hfpv0l) ([raw file](https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/5b51a33652fab0eae869d59ea9c49e08f37870bc/pkg/csi/controller.go#L560)):*
<details><summary><i>Previously, Anirudh9794 wrote…</i></summary><blockquote>

break seems to be redundant here
</blockquote></details>

removed

**Comment by arunmk** [comment]
*[`manifests/csi-controller-crs.yaml.template` line 175 at r3](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-NnQjMWABzyuGe-R6WqK:-NngZ7jHDG8JFT37YKON:btj5c5s) ([raw file](https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/5b51a33652fab0eae869d59ea9c49e08f37870bc/manifests/csi-controller-crs.yaml.template#L175)):*
<details><summary><i>Previously, ymo24 wrote…</i></summary><blockquote>

+1
</blockquote></details>

These should actually be registry.k8s.io itself. The entries that correspond to the air-gapped __REGISTRY__ are in the `artifacts` folder.

**Comment by arunmk** [comment]
*[`manifests/csi-controller.yaml.template` line 175 at r3](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-NnQjWCr2wQz11qtdYIx:-NngZJSBDKUhieU2CJZY:btj5c5s) ([raw file](https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/5b51a33652fab0eae869d59ea9c49e08f37870bc/manifests/csi-controller.yaml.template#L175)):*
<details><summary><i>Previously, Anirudh9794 wrote…</i></summary><blockquote>

This image should have `__REGISTRY__`
</blockquote></details>

These should actually be registry.k8s.io itself. The entries that correspond to the air-gapped __REGISTRY__ are in the `artifacts` folder.

**Comment by arunmk** [comment]
*[`manifests/csi-node-crs.yaml.template` line 76 at r3](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-NnQjieEAl6vnIol-ZsF:-NngZMpl85_PJEYCErDd:btj5c5s) ([raw file](https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/5b51a33652fab0eae869d59ea9c49e08f37870bc/manifests/csi-node-crs.yaml.template#L76)):*
<details><summary><i>Previously, Anirudh9794 wrote…</i></summary><blockquote>

__REGISTRY__ here as well
</blockquote></details>

These should actually be registry.k8s.io itself. The entries that correspond to the air-gapped __REGISTRY__ are in the `artifacts` folder.

**Comment by arunmk** [comment]
*[`manifests/csi-node.yaml.template` line 77 at r3](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-NnQjoIM0zpyFclFRNAS:-NngZPPKBkrUwN5ppZBz:bsjuezg) ([raw file](https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/5b51a33652fab0eae869d59ea9c49e08f37870bc/manifests/csi-node.yaml.template#L77)):*
<details><summary><i>Previously, Anirudh9794 wrote…</i></summary><blockquote>

__REGISTRY__ here as well
</blockquote></details>

These should actually be registry.k8s.io itself. The entries that correspond to the air-gapped \__REGISTRY__ are in the `artifacts` folder.

**Comment by arunmk** [comment]
*[`pkg/csi/controller.go` line 295 at r3](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-NnQmenr9qOerAxWRAv-:-NngZWAp1Jf9GyTBI3gh:b-whgw81) ([raw file](https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/48314b27ba85aff74d25569a149ce5db5d10ee6e/pkg/csi/controller.go#L295)):*
<details><summary><i>Previously, Anirudh9794 wrote…</i></summary><blockquote>

Just wanted to understand why this check was not required. Is it something like if no
</blockquote></details>

Resolved OOB

**Comment by arunmk** [comment]
*[`pkg/csi/controller.go` line 410 at r3](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-NnQo7vx09f1pAIawmir:-Nngd-ZXBuJgTGPXWfWS:b-2iwowi) ([raw file](https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/5b51a33652fab0eae869d59ea9c49e08f37870bc/pkg/csi/controller.go#L410)):*
<details><summary><i>Previously, Anirudh9794 wrote…</i></summary><blockquote>

How can one set the limit bytes in the request?
</blockquote></details>

By using the `limit` parameter in the yaml. Am I understanding the question correctly?

**Review by Anirudh9794 — APPROVED** [review]
> Reviewed 9 of 10 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144)* status: 1656 of 1657 files reviewed, 6 unresolved discussions (waiting on @arunmk, @rocknes, and @ymo24)

___
*[`pkg/csi/controller.go` line 410 at r3](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-NnQo7vx09f1pAIawmir:-NnkYH1G0kKk99NnWHIJ:bvvqqrr) ([raw file](https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/5b51a33652fab0eae869d59ea9c49e08f37870bc/pkg/csi/controller.go#L410)):*
<details><summary><i>Previously, arunmk (Arun M. Krishnakumar) wrote…</i></summary><blockquote>

By using the `limit` parameter in the yaml. Am I understanding the question correctly?
</blockquote></details>

I was wondering which object will have the limit (PV or PVC)
___
*[`pkg/csi/controller.go` line 471 at r3](https://reviewable.io/reviews/vmware/cloud-director-named-disk-csi-driver/144#-NnekWt77J35aAswGnZ0:-NnkYZK58wXp53E5bMyC:bowkwf7) ([raw file](https://github.com/vmware/cloud-director-named-disk-csi-driver/blob/5b51a33652fab0eae869d59ea9c49e08f37870bc/pkg/csi/controller.go#L471)):*
<details><summary><i>Previously, arunmk (Arun M. Krishnakumar) wrote…</i></summary><blockquote>

The default sets `expansionMode = csi.PluginCapability_VolumeExpansion_ONLINE` and then there is a `if` block. So the `else is the default itself.

Am I understanding your comment correctly @Anirudh9794 ?
</blockquote></details>

Yes, sorry. I think I missed that line

**Pull request merged**